PDFExporter/Categories/UIKit/UITableView+PDFExporterDrawing.m

```diff
@@ -134,4 +108,27 @@ - (CGRect)subviewRect:(UIView *)subview layoutPageRect:(CGRect)rect {
     }
 }
 
+- (void)setFrameForContentOffset:(CGPoint)contentOffset {
+    PDFExporter_dispatch_sync_main_queue(^{
+        self.contentOffset = contentOffset;
+        CGRect frame = self.frame;
+        [self layoutIfNeeded];
+        self.frame = frame;
+        [self layoutHeadersAndFooters];
+    });
+}
+
+- (void)computeOffsetForRect:(CGRect)rect additionalCode:(void (^)(void))additionalCode {
+    CGPoint contentOffset = CGPointZero;
+    for (CGFloat yCoordinate = CGRectGetMinY(rect);
+         yCoordinate < CGRectGetMaxY(rect);
+         yCoordinate += CGRectGetHeight(self.bounds)) {
+
+        contentOffset.y = yCoordinate;
+
+        [self setFrameForContentOffset:contentOffset];
+        additionalCode();
```

Comment:
Shall we just execute ```additionalCode``` just where needed outside of method (in the calling function)?

Reply:
This code was executed for each ```yCoordinate``` so I guess that it has to be in the ```for``` loop.

Reply:
@dmacra 
The idea of removing code duplication was to have this loop as a reusable function. The block is good to have because we usually need to do additional work in each scrolling iteration.